### PR TITLE
[FEATURE] More validations on YAML rule editor

### LINKED ISSUE #214: [FEATURE] More inline validations on YAML rule editor
**Is your feature request related to a problem?**
[Update: [Ming's UX improvement points](https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/215#issuecomment-1367681982) included this issue]
Visual editor string character restrictions validation on:
- Rule name
- Description
- Author

Should be shown inlined on yaml editor blur/submit. 

https://user-images.githubusercontent.com/1564068/207247650-b674fc53-a3d4-4120-9716-4489785a9e3c.mov

### PR DESCRIPTION
**Pull request by djindjic**
### Issues Resolved
Closes #214 #207 

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/Rules/components/RuleEditor/RuleEditorForm.tsx [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+                errors={Object.keys(props.errors).map(
+                  (key) => props.errors[key as keyof RuleEditorFormModel] as string
+                )}
+                change={(e) => {
+                  const formState = mapRuleToForm(e);
+                  props.setValues(formState);
+                }}
+              ></YamlRuleEditorComponent>
+            )}
+            <FormSubmitionErrorToastNotification notifications={notifications} />
```
Comment (amsiglan) [inline public/pages/Rules/components/RuleEditor/RuleEditorForm.tsx:145]: Type: `ForSubmissionErrorToastNotification`

File: public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      setState((prevState) => ({ ...prevState, errors: null }));
+    } catch (error) {
+      setState((prevState) => ({ ...prevState, errors: ['Invalid YAML'] }));
+
+      console.warn('Security Analytics - Rule Eritor - Yaml load', error);
+    }
+  };
+
+  const renderErrors = () => {
+    if (state.errors && state.errors.length > 0) {
```
Comment (amsiglan) [inline public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx:67]: rewrite as `state.errors?.length > 0`

File: public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      return (
+        <EuiCallOut size="m" color="danger" title="Please address the highlighted errors.">
+          <ul>
+            {state.errors.map((error, i) => (
+              <li key={i}>{error}</li>
+            ))}
+          </ul>
+        </EuiCallOut>
+      );
+    } else if (isInvalid && errors && errors.length > 0) {
```
Comment (amsiglan) [inline public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx:77]: Rewrite `errors && errors.length > 0` as `errors?.length > 0`

File: public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx (hunk 2 above)
Comment (djindjic) [reply public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx:67]: not possible like this way if I've understood right?
image

Comment (amsiglan) [reply public/pages/Rules/components/RuleEditor/components/YamlRuleEditorComponent/YamlRuleEditorComponent.tsx:67]: aah, leave it as is then

**Review by amsiglan — APPROVED** [review]

**Comment by djindjic** [comment]
@amsiglan I've addressed [Ming's UX improvement points](https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/215#issuecomment-1367681982) in this PR

**Review by sbcd90 — APPROVED** [review]

**Comment by amsiglan** [comment]
The windows cypress test failure was due to a slower loading time and unrelated to the changes made in this PR, hence it was okay to merge the changes.

**Pull request merged**